gamemodes/terrortown/gamemode/shared/sh_util.lua

```diff
@@ -107,27 +119,44 @@ end
 -- @realm shared
 function util.GetNextAlivePlayer(ply)
 	local alive = util.GetAlivePlayers()
-
 	if #alive < 1 then return end
 
-	local prev = nil
-	local choice = nil
-
 	if IsValid(ply) then
-		for _, p in pairs(alive) do
+		local prev = nil
+
+		for i = 1, #alive do
 			if prev == ply then
-				choice = p
+				return alive[i]
 			end
 
-			prev = p
+			prev = alive[i]
 		end
 	end
 
-	if not IsValid(choice) then
-		choice = alive[1]
-	end
+	return alive[1]
+end
+
+---
+-- Returns a random entry of the given @{table}
+-- @param table tbl the @{table} that contains the data
+-- @param nil|function filterFn the @{function} that has to return true on the given entry
+-- @return any the entry that returned true on the given @{function}
+-- @note The given @{table} has to be iterable.
+-- @note The returned entry will get removed from the given @{table}. If you wanna keep the original table untouched, create a copy for this function.
```

Comment:
I would really like this to be a warning or even rename the function because a getter function indicates value retrieval (undistructive) and not value extraction (destructive) imo